File: src/Notification/data/IRepository.ts
export default interface IRepository<TEntity> {
    Single(predicate: any): Promise<TEntity | null | undefined>;
    List(predicate: any, skip: number, lastId: string, take: number): Promise<Array<TEntity> | null | undefined>;
    Insert(entity: TEntity): Promise<TEntity | string>;
    Update(id: string, entity: TEntity): Promise<TEntity | string | null | undefined>;
    Delete(id: string): Promise<TEntity | string | number | null | undefined>;
}

File: src/Account/Data/IRepository.cs
namespace Account.Data;

public interface IRepository<TEntity> where TEntity : IEntity
{
    public TEntity? Single(Func<TEntity, bool>? predicate = null);
    public IEnumerable<TEntity> List(Func<TEntity, bool>? predicate = null, int skip = default, string? lastId = default, int take = default);
    public TEntity Insert(TEntity entity);
    public TEntity? Update(string Id, TEntity entity);
    public TEntity? Delete(string Id);
}